### PR TITLE
[C++][RestSDK] Windows vcpkg vs2017 support

### LINKED ISSUE #2747: [REQ][cpp-restsdk] Better support for vckpg and visual studio 2017
### Is your feature request related to a problem? Please describe.

When generating a cpp-restsdk client, the project produced isn’t easy to compile with visual studio. The suggested command for vcpkg only partially works, and the included folders aren’t setup correctly by default.

## Describe the solution you'd like

Some more hints or support about how to get this working in windows would be helpful for newcomers.

## Describe alternatives you've considered

I've manually edited the CMakeLists.txt file to get this working.

## Additional context

This problem affects users trying to build the cpp-restsdk, on a windows machine, using vckpg, and visual studio 2017.

### PR DESCRIPTION
**Pull request by dan-drl**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`~~, `3.4.x`, `4.0.x`~~. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

This addresses comments made in #2747 (Better support for vckpg and visual studio 2017).

(details of the change, additional tests that have been done, reference to the issue for tracking, etc)

Fixes #2747

### CODE REVIEW

**Comment by etherealjoy** [comment]
@stkrwork @etherealjoy @MartinDelille @fvarose @ravinikam

File: modules/openapi-generator/src/main/resources/cpp-rest-sdk-client/api-header.mustache [hunk 1]
```diff
@@ -15,6 +15,10 @@
 {{/imports}}
 {{^hasModelImport}}#include "../ModelBase.h"{{/hasModelImport}}
 
+#ifdef _WIN32 | _WIN64
```
Comment (etherealjoy) [inline modules/openapi-generator/src/main/resources/cpp-rest-sdk-client/api-header.mustache]: Small nit...
Would it make sense to add this in ApiClient header instead?

Comment (dan-drl) [reply modules/openapi-generator/src/main/resources/cpp-rest-sdk-client/api-header.mustache]: Good call. I just updated my branch accordingly.

**Comment by etherealjoy** [comment]
I got a warning with this during compilation
```
/home/dev/openapi-generator/samples/client/petstore/cpp-restsdk/api/../ApiClient.h:31:14: warning: extra tokens at end of #ifdef directive
 #ifdef _WIN32|_WIN64
```
I think this syntax would be more suitable due to no warning.
```
#if  defined (_WIN32) || defined (_WIN64)
#undef U
#endif
```

**Comment by dan-drl** [comment]
Alright, I also switched my branch to recommended.  #if defined (_WIN32) || defined (_WIN64).

**Pull request merged**